plugins/nf-parquet/build.gradle

```diff
@@ -50,34 +50,26 @@ sourceSets {
 }
 
 ext {
-    nextflowVersion = '23.04.0'
+    nextflowVersion = '24.04.2'
 }
 
 dependencies {
     // This dependency is exported to consumers, that is to say found on their compile classpath.
     compileOnly "io.nextflow:nextflow:$nextflowVersion"
     compileOnly 'org.slf4j:slf4j-api:1.7.10'
     compileOnly 'org.pf4j:pf4j:3.4.1'
+
     // add here plugins depepencies
-    api('org.apache.parquet:parquet-avro:1.10.0')
-    api('org.apache.hadoop:hadoop-common:3.1.0') {
-        exclude group: 'org.slf4j', module: 'slf4j-log4j12' 
-    }
+    implementation 'com.jerolba:carpet-record:0.1.0'
 
     // test configuration
-    testImplementation "org.codehaus.groovy:groovy:3.0.17"
-    testImplementation "org.codehaus.groovy:groovy-nio:3.0.17"
     testImplementation "io.nextflow:nextflow:$nextflowVersion"
-    testImplementation ("org.codehaus.groovy:groovy-test:3.0.17") { exclude group: 'org.codehaus.groovy' }
-    testImplementation ("cglib:cglib-nodep:3.3.0")
-    testImplementation ("org.objenesis:objenesis:3.1")
-    testImplementation ("org.spockframework:spock-core:2.2-groovy-3.0") { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
-    testImplementation ('org.spockframework:spock-junit4:2.2-groovy-3.0') { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
-    testImplementation ('com.google.jimfs:jimfs:1.1')
-
+    testImplementation ("org.spockframework:spock-core:2.2-groovy-4.0") { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
+    testImplementation ('org.spockframework:spock-junit4:2.2-groovy-4.0') { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
```

Comment:
Let's use spock 2.3 as in nextflow core 

https://github.com/nextflow-io/nextflow/blob/5a37e6177f7a0e02b2af922768a0df5984b07b7b/modules/nextflow/build.gradle#L59-L61